### DISCUSSION TITLE
Nested in-memory grid editor problems

### DISCUSSION THREAD

**Discussion by petebr** [serenity-is/Serenity#6581] [Help] [answered]
I love using in-memory grids for quick results but I always have issues when adding a second level to one, eg, Main table A has multiple children B, grid editor and dialog works fine. 

Setup using this construct
```C#
class TableARow
...
 [MasterDetailRelation(foreignKey: "TableAId")]
        public List<TableBRow> Children

```

 If I add a child table to B then things start to fall apart. It seems to save the details at C but never load them back. Then I start seeing `IComparable ` errors until I switch it out for a regular grid built by hand - which is fine but not very elegant when it comes to creating new data - having to hide tabs of child tables until the main record exists.
Is there a restriction there? Anyone else found this issue? 
Would it help to create a repo that describes it?

**Comment by dfaruque** [comment, marked as answer]
use `IncludeColumnNames` parameter like the following 
`[MasterDetailRelation(foreignKey: nameof(DetailRow.MasterId), IncludeColumnNames = new[] { nameof(DetailRow.Detial2ListField) })]`

**Reply by petebr** [reply]
oh cool, that works for loading up the nested detail table - I had previously tried setting `  [MinSelectLevel(SelectLevel.Always)]` hoping that would do the same, but no. I'll certainly use this syntax in future.
However, I'm still getting a `System.ArgumentException: 'At least one object must implement IComparable.'` inside the aftersave behaviour of the inner detail table. All packages up-to-date on net6. I'll try setting up a new repo to reproduce but I assume you're not seeing that error, meanwhile I'll mark you as the answer, 
Thanks.

**Reply by dfaruque** [reply]
try setting `CheckChangesOnUpdate = false`
eg 

`[MasterDetailRelation(foreignKey: nameof(DetailRow.MasterId), CheckChangesOnUpdate = false, IncludeColumnNames = new[] { nameof(DetailRow.Detial2ListField) })]`

**Reply by petebr** [reply]
That worked, thank you. 
I setup a new project that worked fine with a nested grid and I was trying to work out what difference I had introduced but I couldn't tell.
